examples/erc20_gas/src/handler.rs

```diff
@@ -49,77 +49,33 @@ where
     type HaltReason = HaltReason;
 
     fn validate_against_state_and_deduct_caller(&self, evm: &mut Self::Evm) -> Result<(), ERROR> {
-        let context = evm.ctx();
-        let basefee = context.block().basefee() as u128;
-        let blob_price = context.block().blob_gasprice().unwrap_or_default();
-        let is_balance_check_disabled = context.cfg().is_balance_check_disabled();
-        let is_eip3607_disabled = context.cfg().is_eip3607_disabled();
-        let is_nonce_check_disabled = context.cfg().is_nonce_check_disabled();
-        let caller = context.tx().caller();
-        let value = context.tx().value();
+        let (tx, block, cfg, journal) = evm.ctx_mut().tx_block_cfg_journal_mut();
```

Comment:
No need for temp fields, and we can use `tx_block_cfg_journal_mut` to get all needed context objects.